src/lib/site-import/status.ts

```diff
@@ -64,34 +72,40 @@ const IMPORT_SQL_PROGRESS_QUERY = gql`
 	}
 `;
 
-export type ImportSqlCheckStatusInput = {
-	app: Object,
-	env: Object,
-	progressTracker: ProgressTracker,
-};
+export interface ImportSqlCheckStatusInput {
+	app: App;
+	env: AppEnvironment;
+	progressTracker: ProgressTracker;
+}
 
-async function getStatus( api, appId, envId ) {
-	const response = await api.query( {
+interface GetStatusResponse {
+	importStatus?: AppEnvironmentImportStatus | null;
+	importJob?: Job | PrimaryDomainSwitchJob | null;
+	launched?: boolean | null;
+}
+
+async function getStatus(
+	api: ApolloClient< NormalizedCacheObject >,
+	appId: number,
+	envId: number
+): Promise< GetStatusResponse > {
+	const response = await api.query< AppQuery, AppQueryVariables >( {
 		query: IMPORT_SQL_PROGRESS_QUERY,
 		variables: { appId, envId },
 		fetchPolicy: 'network-only',
 	} );
-	const {
-		data: {
-			app: { environments },
-		},
-	} = response;
-	if ( ! environments?.length ) {
+	const environments = response.data.app?.environments ?? [];
+	if ( ! environments.length ) {
 		throw new Error( 'Unable to determine import status from environment' );
 	}
-	const [ environment ] = environments;
+	const environment = environments[ 0 ] ?? {};
 	const { importStatus, jobs, launched } = environment;
 
 	if ( ! environment.isK8sResident && ! jobs?.length ) {
 		return {};
 	}
 
-	const [ importJob ] = jobs;
+	const importJob = jobs?.[ 0 ] ?? {};
```

Comment:
TS complains that `jobs` could be `undefined`. In this case, destructuring will fail.

Reply:
Good catch, fixed!